### PR TITLE
Rubicon Bid Adapter: PAAPI/Fledge support

### PR DESCRIPTION
**Pull request by spotxslagle**
## Type of change

- [ x ] Feature

## Description of change

Rubicon Bid Adapter support for passing PAAPI/Fledge auction configs

### CODE REVIEW

File: modules/rubiconBidAdapter.js [hunk 1]
```diff
@@ -716,6 +720,16 @@ export const spec = {
     }, []).sort((adA, adB) => {
       return (adB.cpm || 0.0) - (adA.cpm || 0.0);
     });
+
+    let fledgeAuctionConfigs = responseObj.component_auction_config?.map(config => {
+      return { config, bidId: config.bidId }
```
Comment (robertrmartinez) [inline modules/rubiconBidAdapter.js:725]: Does this `bidId` returned by AE have to match the `bidId` prebid has for the bidRequest?

Where is DV+ getting this `bidId`?

File: modules/rubiconBidAdapter.js [hunk 2]
```diff
@@ -520,6 +521,10 @@ export const spec = {
     if (configUserId) {
       data['ppuid'] = configUserId;
     }
+
+    if (bidderRequest.fledgeEnabled) {
```
Comment (dgirardi) [inline modules/rubiconBidAdapter.js:525]: I think you should look at `ortb2Imp.ext.ae` instead, the pub can enable it only for some slots.

File: modules/rubiconBidAdapter.js [hunk 3]
```diff
@@ -716,6 +720,16 @@ export const spec = {
     }, []).sort((adA, adB) => {
       return (adB.cpm || 0.0) - (adA.cpm || 0.0);
     });
+
+    let fledgeAuctionConfigs = responseObj.component_auction_config?.map(config => {
+      return { config, bidId: config.bidId }
+    });
+
+    if (fledgeAuctionConfigs) {
```
Comment (dgirardi) [inline modules/rubiconBidAdapter.js:728]: `return {bids, fledgeAuctionConfigs}` should work, core will ignore the second property if it's not an array and still pick up the bids

File: modules/rubiconBidAdapter.js (hunk 1 above)
Comment (spotxslagle) [reply modules/rubiconBidAdapter.js:725]: Yes. They're getting it from the fastlane param we pass in the adapter

File: modules/rubiconBidAdapter.js (hunk 3 above)
Comment (spotxslagle) [reply modules/rubiconBidAdapter.js:728]: While that would be better, it breaks some unit tests it's not worth fixing right now. I'll plan to deal with this when we aren't trying to get this out so quickly

**Comment by spotxslagle** [comment]
@dgirardi would you mind giving this another look? I'm hoping to get this merged today. Thank you!

**Review by dgirardi — APPROVED** [review]
LGTM!

**Review by robertrmartinez — APPROVED** [review]

**Pull request merged**